File: frontend/src/api/family.ts
import api from './index'

export interface Family {
  id: number
  name: string
  is_default: boolean
  created_at: string
  members: FamilyMember[]
}

export interface FamilyMember {
  id: number
  user_id: number
  username: string
  role: string
  joined_at: string
}

export interface FamilyCreate {
  name: string
}

export interface AddMemberRequest {
  username: string
  role?: string
}

export const familyApi = {
  getAll: async (): Promise<Family[]> => {
    const response = await api.get<{ families: Family[] }>('/families')
    return response.data.families
  },

  getById: async (id: number): Promise<Family> => {
    const response = await api.get<Family>(`/families/${id}`)
    return response.data
  },

  create: async (data: FamilyCreate): Promise<Family> => {
    const response = await api.post<Family>('/families', data)
    return response.data
  },

  update: async (id: number, data: FamilyCreate): Promise<Family> => {
    const response = await api.put<Family>(`/families/${id}`, data)
    return response.data
  },

  delete: async (id: number): Promise<void> => {
    await api.delete(`/families/${id}`)
  },

  addMember: async (familyId: number, data: AddMemberRequest): Promise<FamilyMember> => {
    const response = await api.post<FamilyMember>(`/families/${familyId}/members`, data)
    return response.data
  },

  removeMember: async (familyId: number, userId: number): Promise<void> => {
    await api.delete(`/families/${familyId}/members/${userId}`)
  }
}

File: backend/app/schemas/family.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class FamilyBase(BaseModel):
    name: str


class FamilyCreate(FamilyBase):
    pass


class FamilyUpdate(FamilyBase):
    name: Optional[str] = None


class FamilyMemberResponse(BaseModel):
    id: int
    user_id: int
    username: str
    role: str
    joined_at: datetime

    class Config:
        from_attributes = True


class FamilyResponse(FamilyBase):
    id: int
    is_default: bool
    created_at: datetime
    members: list[FamilyMemberResponse] = []

    class Config:
        from_attributes = True


class FamilyListResponse(BaseModel):
    families: list[FamilyResponse]


class AddMemberRequest(BaseModel):
    username: str
    role: str = "member"  # admin, member
